Userland/Libraries/LibWeb/HTML/Worker.cpp

```diff
@@ -0,0 +1,363 @@
+
+                // 2. Fetch request, and asynchronously wait to run the remaining steps
+                //    as part of fetch's process response for the response response.
+                // Implied
+
+                // 3. Set worker global scope's url to response's url.
+                //worker_global_scope->set_url(url);
+
+                // 4. Initialize worker global scope's policy container given worker global scope, response, and inside settings.
+                // FIXME: implement policy container
+
+                // 5. If the Run CSP initialization for a global object algorithm returns "Blocked" when executed upon worker
+                //    global scope, set response to a network error. [CSP]
+                // FIXME: CSP support
+
+                // 6. If worker global scope's embedder policy's value is compatible with cross-origin isolation and is shared is true,
+                //    then set agent's agent cluster's cross-origin isolation mode to "logical" or "concrete".
+                //    The one chosen is implementation-defined.
+                // FIXME: CORS policy support
+
+                // 7. If the result of checking a global object's embedder policy with worker global scope, outside settings,
+                //    and response is false, then set response to a network error.
+                // FIXME: Embed policy support
+
+                // 8. Set worker global scope's cross-origin isolated capability to true if agent's agent cluster's cross-origin
+                //    isolation mode is "concrete".
+                // FIXME: CORS policy support
+
+                if (!is_shared) {
+                    // 9.  If is shared is false and owner's cross-origin isolated capability is false, then set worker
+                    //     global scope's cross-origin isolated capability to false.
+                    // 10. If is shared is false and response's url's scheme is "data", then set worker global scope's
+                    //     cross-origin isolated capability to false.
+                }
+            }
+
+            if (data.is_null()) {
+                dbgln_if(WEB_WORKER_DEBUG, "WebWorker: Failed to load {}", url);
+                return;
+            }
+
+            // Asynchronously complete the perform the fetch steps with response.
+            dbgln_if(WEB_WORKER_DEBUG, "WebWorker: Script ready!");
+            auto script = ClassicScript::create(url.to_string(), data, *m_inner_settings, AK::URL());
+
+            // NOTE: Steps 15-31 pick up after step 14 in the main context, steps 1-10 above
+            //       are mostly validation of the response befor
+
+            // 15. Associate worker with worker global scope.
+            // FIXME: Global scope association
+
+            // 16. Let inside port be a new MessagePort object in inside settings's Realm.
+            m_inside_port = MessagePort::create();
+            auto inside_port = m_inside_port;
```

Comment:
Why take a copy of `m_inside_port` here? Can we not just use `auto&` here?

Reply:
If I recall the message port was getting freed and causing a UAF, but I'm not 100% on that. Will try it without and see what I get.

Reply:
Okay, this did work fine not taking a copy of it.